### PR TITLE
[DataFrame API] Add tutorial notebook

### PR DESCRIPTION
**Pull request by caisq**
* Motivation for features / changes
  * Add documentation for the experimental DataFrame API

### CODE REVIEW

File: docs/dataframe_api.ipynb [hunk 1]
```diff
@@ -0,0 +1,678 @@
+{
```
Comment (bileschi) [inline docs/dataframe_api.ipynb:1]: <p>nit: experimnt is spelled wrong.</p><p> </p><p>Also, can you add a name and description to the demo tensorboard, so as demonstrate our idealized use case?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3744/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (bileschi) [inline docs/dataframe_api.ipynb:1]: <p>Suggest replacing "subdirectories" with "a subdirectory".</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3744/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (bileschi) [inline docs/dataframe_api.ipynb:1]: <p>Suggest to add titles to these plots.  Left is "accuracy", Right is "loss"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3744/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

**Comment by bileschi** [comment]
Made a couple of small suggestions using ReviewNB.  Looks great! LGTM

File: docs/dataframe_api.ipynb (hunk 1 above)
Comment (caisq) [reply docs/dataframe_api.ipynb:1]: <p>The typo is fixed. </p><p>Done adding experiment name and description.</p>

Comment (caisq) [reply docs/dataframe_api.ipynb:1]: <p>Done.</p>

Comment (caisq) [reply docs/dataframe_api.ipynb:1]: <p>Done adding titles.</p>

**Comment by caisq** [comment]
<p>Thanks for the review, @bileschi</p>

File: docs/dataframe_api.ipynb [hunk 2]
```diff
@@ -0,0 +1,676 @@
...
```
Comment (GalOshri) [inline docs/dataframe_api.ipynb:1]: <p>What do you think about something like "Accessing TensorBoard data as a DataFrame"?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3744/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (GalOshri) [inline docs/dataframe_api.ipynb:1]: <p>"A feature is introduced in TensorBoard v2.3 supports this use case" -&gt; "TensorBoard 2.3 supports this use case with tensorboard.data.experimental.ExperimentFromDev()"</p><p> </p><p>I don't know if we actually need to mention the version in the text (you check the version later in the notebook anyway). But maybe full information is better.</p><p> </p><p>Consider linking the first mention of "scalar logs" to the appropriate tutorial: <a href="https://www.tensorflow.org/tensorboard/scalars_and_keras" target="_blank">https://www.tensorflow.org/tensorboard/scalars_and_keras</a>. Some people might not think of the word Scalars for this.</p><p> </p><p>The first mention of TensorBoard.dev in note 2 should be a link to TensorBoard.dev.</p><p> </p><p>"filesystem to tensorboard.dev ..." -&gt; "filesystem to TensorBoard.dev"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3744/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (GalOshri) [inline docs/dataframe_api.ipynb:1]: <p>I think we can drop "to make the API usable".</p><p> </p><p>"If you intend to make custom plots ..." -&gt; "We'll use matplotlib and seaborn for custom plots in this guide, but you can use your preferred tool". (the current text makes it sound like matplotlib and seaborn are required to make custom plots).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3744/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (GalOshri) [inline docs/dataframe_api.ipynb:1]: <p>I don't think the comment in the code is necessary here (it is duplicating the text above it)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3744/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (GalOshri) [inline docs/dataframe_api.ipynb:1]: <p>I'd suggest saying something about 'tag' being the metric name. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3744/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (GalOshri) [inline docs/dataframe_api.ipynb:1]: <p>I'd suggest "is not supported by TB's main GUI currently" -&gt; "is not yet supported in TensorBoard's GUI"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3744/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Comment (GalOshri) [inline docs/dataframe_api.ipynb:1]: <p>I'd suggest moving the first comment to be a text cell (it is explaining the next part of the notebook).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3744/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

File: docs/_book.yaml [hunk 3]
```diff
@@ -33,6 +33,8 @@ upper_tabs:
         path: /tensorboard/tensorboard_profiling_keras
       - title: "TensorBoard in notebooks"
         path: /tensorboard/tensorboard_in_notebooks
+      - title: "TensorBoard DataFrame API"
```
Comment (GalOshri) [inline docs/_book.yaml:36]: Thoughts on "TensorBoard data as DataFrames"?

Comment (caisq) [reply docs/_book.yaml:36]: Done.

File: docs/dataframe_api.ipynb (hunk 2 above)
Comment (caisq) [reply docs/dataframe_api.ipynb:1]: Done.

Comment (caisq) [reply docs/dataframe_api.ipynb:1]: Done. Yep, let's be detailed and specific about version.

Comment (caisq) [reply docs/dataframe_api.ipynb:1]: Done.

Comment (caisq) [reply docs/dataframe_api.ipynb:1]: Done.

Comment (caisq) [reply docs/dataframe_api.ipynb:1]: Done.

Comment (caisq) [reply docs/dataframe_api.ipynb:1]: Done.

Comment (caisq) [reply docs/dataframe_api.ipynb:1]: Done.

**Comment by caisq** [comment]
@wchargin let me know if you'd like to review this PR. If I don't hear from you by EOD today, I'll merge it. In that case feel free to comment afterwards nonetheless.

**Pull request merged**